plugins/solidigm/solidigm-internal-logs.c

```diff
@@ -628,23 +627,25 @@ int solidigm_get_internal_log(int argc, char **argv, struct command *command,
 
 	if (log_count > 0) {
 		int ret_cmd;
-		char cmd[ARG_MAX];
+		char *cmd;
 		char *where_err = cfg.verbose ? "" : ">/dev/null 2>&1";
 
 		snprintf(zip_name, sizeof(zip_name), "%s.zip", cfg.dir_prefix);
-		snprintf(cmd, sizeof(cmd), "cd \"%s\" && zip -r \"../%s\" ./* %s", cfg.dir_prefix,
+		asprintf(&cmd, "cd \"%s\" && zip -r \"../%s\" ./* %s", cfg.dir_prefix,
```

Comment:
would be nice to check the return value -1 for asprintf() to make sure malloc internally worked and no other errors.

Reply:
Updated, though I could also see not doing that since the prior code didn't check to see if snprintf overflowed...
Is the update good or too ugly?

Reply:
thanks